src/main/java/fr/zcraft/quartzlib/components/nbt/NBTCompound.java

```diff
@@ -219,7 +221,36 @@ public boolean containsValue(Object value) {
 
     @Override
     public Object put(String key, Object value) {
-        return NBT.toNativeValue(getNbtMap().put(key, NBT.fromNativeValue(value)));
+        try {
+            switch (NBT.fromNativeValue(value).getClass().getName()) {
+                case "net.minecraft.nbt.NBTTagInt":
+                    Method method;
+                    try {
+                        //Reflection.call(nmsNbtTag, "setInt", key, Integer.parseInt(value.toString()));
+                        //Can't use call here because int is casted as an integer
```

Comment:
```suggestion

                        // Cannot use Reflection.call here because int is casted as an integer and we need the method with an int
```

Reply:
Yeah that's a better way to phrase it.